internal/idtokenvalidator/idtokenvalidator_test.go

```diff
@@ -0,0 +1,559 @@
+		assert.NoError(t, err)
+
+		args := defaultJWTArgs
+		args.clientSecret = jwtClientSecret
+		args.payload = map[string]interface{}{
+			"exp": time.Now().Add(-10 * time.Second),
+		}
+
+		token, _, err := givenAJWT(t, args)
+		assert.NoError(t, err)
+
+		err = validator.Validate(token, OptionalVerification{})
+		assert.ErrorContains(t, err, "\"exp\" not satisfied")
+	})
+
+	t.Run("allows clock skew on exp", func(t *testing.T) {
+		validator, err := New(context.Background(), jwtDomain, jwtClientID, jwtClientSecret, "HS256", WithClockTolerance(10*time.Second))
+		assert.NoError(t, err)
+
+		args := defaultJWTArgs
+		args.clientSecret = jwtClientSecret
+		args.payload = map[string]interface{}{
+			"exp": time.Now().Add(-5 * time.Second),
+		}
+
+		token, _, err := givenAJWT(t, args)
+		assert.NoError(t, err)
+
+		err = validator.Validate(token, OptionalVerification{})
+		assert.NoError(t, err)
+	})
+
+	t.Run("passes when auth_time is within max_age", func(t *testing.T) {
+		validator, err := New(context.Background(), jwtDomain, jwtClientID, jwtClientSecret, "HS256")
+		assert.NoError(t, err)
+
+		args := defaultJWTArgs
+		args.clientSecret = jwtClientSecret
+		args.payload = map[string]interface{}{
+			"auth_time": time.Now().Add(-100 * time.Second).Unix(),
+		}
+
+		token, _, err := givenAJWT(t, args)
+		assert.NoError(t, err)
+
+		err = validator.Validate(token, OptionalVerification{MaxAge: 200 * time.Second})
+		assert.NoError(t, err)
+	})
+
+	t.Run("errors when auth_time is beyond max_age", func(t *testing.T) {
+		validator, err := New(context.Background(), jwtDomain, jwtClientID, jwtClientSecret, "HS256")
+		assert.NoError(t, err)
+
+		args := defaultJWTArgs
+		args.clientSecret = jwtClientSecret
+		args.payload = map[string]interface{}{
+			"auth_time": time.Now().Add(-500 * time.Second).Unix(),
+		}
+
+		token, _, err := givenAJWT(t, args)
+		assert.NoError(t, err)
+
+		err = validator.Validate(token, OptionalVerification{MaxAge: 200 * time.Second})
+		assert.ErrorContains(t, err, "auth_time claim in the ID token indicates that too much time has passed since")
+	})
+
+	t.Run("allows clock skew on auth_time", func(t *testing.T) {
+		validator, err := New(context.Background(), jwtDomain, jwtClientID, jwtClientSecret, "HS256", WithClockTolerance(100*time.Second))
+		assert.NoError(t, err)
+
+		args := defaultJWTArgs
+		args.clientSecret = jwtClientSecret
+		args.payload = map[string]interface{}{
+			"auth_time": time.Now().Add(-250 * time.Second).Unix(),
+		}
+
+		token, _, err := givenAJWT(t, args)
+		assert.NoError(t, err)
+
+		err = validator.Validate(token, OptionalVerification{MaxAge: 200 * time.Second})
+		assert.NoError(t, err)
+	})
+
+	t.Run("verifies auth_time is a number that exists when max_age is passed", func(t *testing.T) {
+		validator, err := New(context.Background(), jwtDomain, jwtClientID, jwtClientSecret, "HS256")
+		assert.NoError(t, err)
+
+		args := defaultJWTArgs
+		args.clientSecret = jwtClientSecret
```

Comment:
Should we add `args.payload = map[string]interface{}{}` here to make the absence of `auth_time` clearer?